components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -161,15 +153,35 @@ class Nimbus : NimbusApi {
 
     override fun updateExperiments() {
         if (!isInitialized) return
-        nimbus.updateExperiments()
-        onExperimentUpdated?.invoke(nimbus.getActiveExperiments())
+        try {
+            // We need to launch this off of the main thread because it involves a network request
+            // within the nimbus-sdk.
+            scope.launch {
+                nimbus.updateExperiments()
+                experimentsUpdated()
+            }
+        } catch (e: ErrorException.RequestError) {
+            logger.info("Error fetching experiments from endpoint: $e")
+        } catch (e: ErrorException.InvalidExperimentResponse) {
+            logger.info("Invalid experiment response: $e")
+        }
```

Comment:
Wrapping a try-catch around a `scope.launch {}` is probably fine, but it looks weird. Suggest putting it the try-catch within the launch block.